### ISSUE TITLE
Failed to build protobuf-gen-c as a part of a bigger CMake project

### ISSUE THREAD

**Issue by KivApple** [protobuf-c/protobuf-c#480]
I am trying to build protobuf-gen-c as a build step of my own project building to avoid require install protobuf on the build system:

```
set(BUILD_SHARED_LIBS OFF)
set(protobuf_BUILD_TESTS OFF)
add_subdirectory(lib/protobuf/cmake "${CMAKE_CURRENT_BINARY_DIR}/protobuf")

set(Protobuf_INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}/lib/protobuf/src")
set(Protobuf_LIBRARIES libprotobuf)
set(Protobuf_PROTOC_LIBRARY libprotoc)
if(MSVC)
	set(
			Protobuf_PROTOC_EXECUTABLE
			"${CMAKE_CURRENT_BINARY_DIR}/protobuf/${CMAKE_BUILD_TYPE}/protoc${CMAKE_EXECUTABLE_SUFFIX}"
			-I${Protobuf_INCLUDE_DIR}
	)
else()
	set(
			Protobuf_PROTOC_EXECUTABLE
			"${CMAKE_CURRENT_BINARY_DIR}/protobuf/protoc${CMAKE_EXECUTABLE_SUFFIX}"
			-I${Protobuf_INCLUDE_DIR}
	)
endif()

add_subdirectory(lib/protobuf-c/build-cmake "${CMAKE_CURRENT_BINARY_DIR}/protobuf-c")
add_dependencies(protoc-gen-c protoc)
```
But I got an error that "protobuf-c.pb.cc" not found (when the building process reaches proto-gen-c compiling).
````
ADD_CUSTOM_COMMAND(OUTPUT protobuf-c/protobuf-c.pb.cc protobuf-c/protobuf-c.pb.h
                   COMMAND ${PROTOBUF_PROTOC_EXECUTABLE}
                   ARGS --cpp_out ${CMAKE_BINARY_DIR} -I${MAIN_DIR} ${MAIN_DIR}/protobuf-c/protobuf-c.proto)
FILE(GLOB PROTOC_GEN_C_SRC ${MAIN_DIR}/protoc-c/*.h ${MAIN_DIR}/protoc-c/*.cc )
ADD_EXECUTABLE(protoc-gen-c ${PROTOC_GEN_C_SRC} protobuf-c/protobuf-c.pb.cc protobuf-c/protobuf-c.pb.h)
````
This caused because of these lines in build-cmake/CMakeLists.txt. Because ADD_CUSTOM_COMMAND supposed to generate files protobuf-c/protobuf-c.pb.cc and protobuf-c/protobuf-c.pb.h in the current build directory, but actually in this setup it generates files protobuf-c.pb.cc and protobuf-c.pb.h just in the current build directory without "protobuf-c" subfolder.

**Comment by KivApple** [comment]
I've found an error: it is better to use CMAKE_CURRENT_BINARY_DIR instead of CMAKE_BINARY_DIR to support project building via add_subdirectory.

**Comment by edmonds** [comment]
This is probably fixed in 1.5.0, please reopen if that's not the case.
